viziphant/unitary_event_analysis.py

```diff
@@ -209,12 +186,15 @@ def mark_epochs(axes_name):
         """
         for key in params_dict['events'].keys():
             for event_timepoint in params_dict['events'][key]:
-                axes_name.axvline(event_timepoint, ls='-',
-                                  lw=params_dict['lw'], color='r')
-                if axes_name.get_geometry()[2] == 6:
-                    axes_name.text(x=event_timepoint-10*pq.ms,
-                                   y=-38, s=key,
-                                   fontsize=12, color='b')
+                # check if epochs are between time-axis limits
+                if ((xlim_left <= event_timepoint) and
+                        (event_timepoint <= xlim_right)):
+                    axes_name.axvline(event_timepoint, ls='-',
+                                      lw=params_dict['lw'], color='r')
+                    if axes_name.get_geometry()[2] == 6:
+                        axes_name.text(x=event_timepoint-10*pq.ms,
+                                       y=-54, s=key,
+                                       fontsize=12, color='r')
```

Comment:
You do not have to hardcode an offset for the event labels. Matplotlib has options for this:
```suggestion
                        axes_name.text(x=event_timepoint,
                                       y=-54, s=key,
                                       fontsize=12, color='r',
                                       horizontalalignment='center')
```
`horizontalalignment` takes values `'left'`, `'center'`, `'right'`.  A demo can be found [here](https://matplotlib.org/3.1.1/gallery/text_labels_and_annotations/text_alignment.html#sphx-glr-gallery-text-labels-and-annotations-text-alignment-py).